uploader/download.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+const downloader = require("./downloader");
+
+module.exports = function(version) {
+  const platforms = [ "win", "mac", "linux" ];
+  const architectures = [ "x86", "x64" ];
+
+  const downloads = flatMap(platforms.map(function(platform) {
+    return architectures.map(function(arch) {
+      return downloader(version, platform, arch);
+    })
+  }));
+
+  return Promise.all(downloads)
+    .then(function(downloads) {
+      /*
+       * If a version couldn't be fetched (probably because of a 404)
+       * then there will be nulls in the array.
+       */
+      return downloads.filter(function(download) {
+        return download != null;
+      });
+    })
+    .catch(function(err) {
+      console.error(err, err.stack);
+    });
+};
+
+function flatMap(arr) {
```

Comment:
It's a bit confusing calling this `flatMap` as there's no mapping going on here. Just call it `flatten` perhaps.

Reply:
Fair enough, it's a hangover from some other code I was mucking around with.